src/mds/MDSNotificationManager.h

```diff
@@ -1,29 +1,36 @@
 #pragma once
-
 #include "CDentry.h"
 #include "CInode.h"
+#include "MDSRank.h"
+#include "common/ceph_context.h"
+#include "include/buffer.h"
+#include <bits/stdc++.h>
 
 #ifdef WITH_CEPHFS_NOTIFICATION
 #include "MDSKafka.h"
 #include "MDSNotificationMessage.h"
 #include "MDSUDPEndpoint.h"
-#endif
 
-#include "common/ceph_context.h"
-#include "include/buffer.h"
-#include <bits/stdc++.h>
+class MDSKafkaManager;
+class MDSUDPManager;
+#endif
 
 class MDSNotificationManager {
 public:
-  MDSNotificationManager(CephContext *cct);
+  MDSNotificationManager(MDSRank *mds);
+  void init();
 
 #ifdef WITH_CEPHFS_NOTIFICATION
```

Comment:
IMO the methods below better to be declared irrespective to WITH_CEPHFS_NOTIFICATION.
But their definition should be empty (or return ENOTSUPP) when WITH_CEPHFS_NOTIFICATION is undefined.
This way one can avoid corresponding #ifdef-s in the core MDS code and keep this WITH/WITHOUT logic inside new modules.